rabbitmq/src/lib.rs

```diff
@@ -48,11 +49,20 @@ impl QueueHandler {
     }
 
     pub async fn setup(&mut self) -> Result<(), Box<dyn std::error::Error>> {
-        let addr = env::var("RABBITMQ_HOST").expect("RABBITMQ_HOST must be set");
-        let port = env::var("RABBITMQ_PORT")
-            .unwrap_or_else(|_| "5672".to_string())
+        let endpoint = env::var("RABBITMQ_ENDPOINT").expect("RABBITMQ_ENDPOINT must be set");
+        let parsed_url = Url::parse(&endpoint).expect("Invalid URL format for RABBITMQ_ENDPOINT");
+
+        let addr = parsed_url
+            .host_str()
+            .expect("RABBITMQ_ENDPOINT must contain a host");
+
+        let port = parsed_url
+            .port()
+            .unwrap_or(5672)
+            .to_string()
             .parse::<u16>()
             .expect("RABBITMQ_PORT must be a valid port number");
```

Comment:
```suggestion
            .unwrap_or(5672);
```

No need to format to string just to parse it back to int.

If the problem is that we need u16 and port from Url isn't u16, use `try_into()`. But I'd expect that it is u16 already.